library/src/main/java/org/xmtp/android/library/Contacts.kt

```diff
@@ -0,0 +1,35 @@
+package org.xmtp.android.library
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.xmtp.android.library.messages.ContactBundle
+import org.xmtp.android.library.messages.ContactBundleBuilder
+import org.xmtp.android.library.messages.Topic
+
+data class Contacts(
+    var client: Client,
+    var knownBundles: Map<String, ContactBundle> = mapOf(),
+    var hasIntroduced: Map<String, Boolean> = mapOf()
+) {
+
+    fun has(peerAddress: String): Boolean =
+        knownBundles[peerAddress] != null
+
+    fun needsIntroduction(peerAddress: String): Boolean =
+        hasIntroduced[peerAddress] != true
+
+    fun find(peerAddress: String): ContactBundle? {
+        val knownBundle = knownBundles[peerAddress]
+        if (knownBundle != null) {
+            return knownBundle
+        }
+        val response =
+            runBlocking(Dispatchers.IO) { client.query(topics = listOf(Topic.contact(peerAddress))) }
```

Comment:
Ooooh I think I misunderstood what `runBlocking` does above. Looks like it can be called from the main thread? If so, maybe we can specify `Dispatchers.IO` for `Client.create()` as well?